src/remote-config/remote-config-api-client.ts

```diff
@@ -125,6 +134,42 @@ export class RemoteConfigApiClient {
       });
   }
 
+  public validateTemplate(template: RemoteConfigContent): Promise<RemoteConfigContent> {
+    return this.getUrl()
+      .then((url) => {
+        const headers = FIREBASE_REMOTE_CONFIG_PUT_HEADERS;
+        headers["If-Match"] = template.etag;
+        const request: HttpRequestConfig = {
+          method: 'PUT',
+          url: `${url}/remoteConfig?validate_only=true`,
+          headers: headers,
+          data: {
+            conditions: template.conditions,
+            parameters: template.parameters,
+          }
+        };
+        return this.httpClient.send(request);
+      })
+      .then((resp) => {
+        if (!Object.prototype.hasOwnProperty.call(resp.headers, 'etag')) {
```

Comment:
I think just checking `validator.isNonEmptyString(resp.headers['etag'])` is enough here?

Reply:
Wanted to differentiate the checks for `property exists` and `is a string`. I think `isNonEmptyString` makes sense, too. Is one preferred over the other?

Reply:
Given that this shouldn't occur in normal conditions, I prefer the approach that requires less typing (i.e. isNonEmptyString).